aggregate/aggregate

```diff
@@ -0,0 +1,223 @@
+        }, inplace=True)
+
+        # Group by relevant columns and aggregate PDB_IDs:
+        string_df = string_df.groupby([
+            "target uniprot gene", 
+            "target uniprot id", 
+            "interactor uniprot gene", 
+            "interactor uniprot id", 
+            "string score"
+        ], as_index=False).agg({
+        "PDB_ID": lambda x: ",".join(sorted(set(map(str, filter(None, x)))))
+        })
+
+        string_df.sort_values(by="string score", ascending=False, inplace=True)
+
+    
+    if not string_df.empty and not mentha_df.empty:
+
+        mentha_target_id = mentha_df["target uniprot id"].iloc[0]
+        string_target_id = string_df["target uniprot id"].iloc[0]
+        if mentha_target_id != string_target_id:
+            print(f"Error: Target Uniprot IDs do not match between mentha and string outputs.")
+            exit(1)
+
+        ## MERGING ##
+        # Merge mentha_df and string_df:
+        merged_df = pd.merge(
+            mentha_df, 
+            string_df[["interactor uniprot id", "PDB_ID", "string score"]], 
+            on="interactor uniprot id", 
+            how="left"
+        )
+
+        # Update structure column:
+        merged_df["structure"] = merged_df.apply(
+            lambda row: ",".join(filter(None, sorted(set(row["structure"].split(",") + row["PDB_ID"].split(",")))))
+            if pd.notna(row["PDB_ID"]) else row["structure"],
+            axis=1
+        )
+        
+        # Drop unnecessary columns:
+        merged_df.drop(columns=[ "PDB_ID"], inplace=True)
+
+        merged_df.sort_values(by=["mentha score", "string score"], ascending=[False, False], inplace=True)
+
+        # Find string entries not in mentha:
+        unmatched_string_df = string_df[~string_df["interactor uniprot id"].isin(mentha_df["interactor uniprot id"])]
+        unmatched_string_df = unmatched_string_df.rename(columns={"PDB_ID": "structure"})
+        unmatched_string_df["mentha score"] = None
+        unmatched_string_df = unmatched_string_df[[
+            "target uniprot id", "target uniprot gene", "interactor uniprot id", 
+            "interactor uniprot gene", "mentha score", "string score", "structure"
+        ]]
+
+        # Concatenate the unmatched string entries to df:
+        final_df = pd.concat([merged_df, unmatched_string_df], ignore_index=True)
+
+    elif mentha_df.empty and not string_df.empty:
+        print(f"Warning: No results found in Mentha for the query protein")
+        final_df = string_df
+        # Add mentha score for consistency:
+        final_df["mentha score"] = ""
+
+    elif string_df.empty and not mentha_df.empty:
+        print(f"Warning: No results found in STRING for the query protein")
+        final_df = mentha_df
+        # Add string score for consistency:
+        final_df["string score"] = ""
+
+    else:
+        print(f"Warning:No results found in Mentha and STRING for the query protein")
+        exit(0)
```

Comment:
I think we could improve the code structure a bit, I don't love doing different things for different states of the input files, I think it would be better to generalize the approach:
  - if mentha or string are empty, then they should be processed in the same way as they were full and result in an empty dataframe, is there any special reason we process them differently? By empty I mean they only have a header (a header is always necessary though)
  - Based on this, the dataframe merge should also work if either mentha or string are empty
  - If both of them are empty, the output should be an empty dataframe respect to erroring out - this is useful if we want to add this to a snakemake pipeline that requires this as output (but it's good to print a warning as well)

so we can probably just get rid of the case-by-case handling